clang/lib/AST/ASTContext.cpp

```diff
@@ -15029,6 +15029,10 @@ static SYCLKernelInfo BuildSYCLKernelInfo(ASTContext &Context,
   MC->mangleCanonicalTypeName(KernelNameType, Out);
   std::string KernelName = Out.str();
 
+  // FIXME: Diagnose kernel names that are not representable in the ordinary
+  // literal encoding. This is not necessarily the right place to add such
+  // a diagnostic.
```

Comment:
@Fznamznon, I'll take care of the character encoding issues. I'm not sure there is actually anything to be done.

Reply:
Character encoding issues have been addressed without additional changes; I just added tests to validate handling of non-ASCII characters in the names of kernel name types.